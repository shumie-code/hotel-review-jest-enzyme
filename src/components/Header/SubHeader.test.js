import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SubHeader from './SubHeader';

describe('the <SubHeader /> component', () => {
    const renderer = new ShallowRenderer();

    it('should render', () => {
        renderer.render(<SubHeader />);
        const component = renderer.getRenderOutput();

        expect(component).toMatchSnapshot();
    });

    it('should render with a dynamic title', () => {
        renderer.render(<SubHeader title='Test Application Test' />);
        const component = renderer.getRenderOutput();

        expect(component).toMatchSnapshot();
    });

    it('should render with a goBack button', () => {
        renderer.render(<SubHeader goBack={() => {}} />);
        const component = renderer.getRenderOutput();

        expect(component).toMatchSnapshot();
    });

    it('should render with a form button', () => {
        renderer.render(<SubHeader openForm={() => {}} />);
        const component = renderer.getRenderOutput();

        expect(component).toMatchSnapshot();
    });
});

