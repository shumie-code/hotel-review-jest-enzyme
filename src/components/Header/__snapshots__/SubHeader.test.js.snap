// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the <SubHeader /> component should render 1`] = `
<ForwardRef(styled.div)>
  <ForwardRef(styled.h2) />
</ForwardRef(styled.div)>
`;

exports[`the <SubHeader /> component should render with a dynamic title 1`] = `
<ForwardRef(styled.div)>
  <ForwardRef(styled.h2)>
    Test Application Test
  </ForwardRef(styled.h2)>
</ForwardRef(styled.div)>
`;

exports[`the <SubHeader /> component should render with a form button 1`] = `
<ForwardRef(styled.div)>
  <ForwardRef(styled.h2) />
  <ForwardRef(Styled(Button))
    onClick={[Function]}
  >
    + Add Review
  </ForwardRef(Styled(Button))>
</ForwardRef(styled.div)>
`;

exports[`the <SubHeader /> component should render with a goBack button 1`] = `
<ForwardRef(styled.div)>
  <ForwardRef(Styled(Button))
    onClick={[Function]}
  >
    &lt; Go Back
  </ForwardRef(Styled(Button))>
  <ForwardRef(styled.h2) />
</ForwardRef(styled.div)>
`;
